applications/FluidDynamicsApplication/custom_elements/symbolic_stokes.cpp

```diff
@@ -198,6 +198,85 @@ void SymbolicStokes<TElementData>::AddBoundaryTraction(
     noalias(rRHS) += prod(rData.lhs,values);
 }
 
+template <class TElementData>
+void SymbolicStokes<TElementData>::Calculate(
+    const Variable<double> &rVariable,
+    double &rOutput,
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+    rOutput = 0.0;
+
+    if (rVariable == HEAT_FLUX) {
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+        rOutput = 0.0;
+        // Shape functions
+        const GeometryType &r_geometry = this->GetGeometry();
+        const unsigned int num_nodes = r_geometry.PointsNumber();
+        Vector data_N(num_nodes);
+        for (unsigned int i = 0; i < num_nodes; i++) {
+            data_N[i] = 1.0 / (double)num_nodes;
+        }
+        data.N = data_N;
+        // Shape functions gradients
+        const GeometryData::IntegrationMethod integration_method = GeometryData::GI_GAUSS_2;
+        ShapeFunctionDerivativesArrayType shape_derivatives;
+        Vector DetJ;
+        r_geometry.ShapeFunctionsIntegrationPointsGradients(shape_derivatives, DetJ, integration_method);
+        data.DN_DX = shape_derivatives[0]; // Note: Valid only for linear elements
+        // Compute strain and stress
+        this->CalculateMaterialResponse(data);
+        // Add limits to the strain, and recompute stress if required
+        const auto &constitutive_law = this->GetConstitutiveLaw();
+        auto strain = data.StrainRate;

```

Comment:
Why creating a copy?

Reply:
because this will help us avoid change the value of strain rate in `data` (in case `ratio` is not equal to 1.)

Reply:
See answer below, this would be no longer needed if computed through the constitutive law.